src/EditorFeatures/CSharpTest/Diagnostics/RemoveNewModifier/RemoveNewModifierCodeFixProviderTests.cs

```diff
@@ -0,0 +1,101 @@
+using Microsoft.CodeAnalysis.CSharp.CodeFixes.RemoveNewModifier;
+using Microsoft.CodeAnalysis.Diagnostics;
+using Microsoft.CodeAnalysis.Test.Utilities;
+using Xunit;
+
+namespace Microsoft.CodeAnalysis.Editor.CSharp.UnitTests.Diagnostics.RemoveNewModifier
+{
+    [Trait(Traits.Feature, Traits.Features.CodeActionsRemoveNewModifier)]
+    public class RemoveNewModifierCodeFixProviderTests : AbstractCSharpDiagnosticProviderBasedUserDiagnosticTest
+    {
+        internal override (DiagnosticAnalyzer, CodeFixProvider) CreateDiagnosticProviderAndFixer(Workspace workspace) =>
+            (null, new RemoveNewModifierCodeFixProvider());
+
+        [Theory]
+        [InlineData(
+            @"public static new void [|Method()|] { }",
+            @"public static void [|Method()|] { }")]
+        [InlineData(
+            "public new int [|Test|];",
+            "public int [|Test|];")]
+        [InlineData(
+            "public new int [|Test|] { get; set; }",
+            "public int [|Test|] { get; set; }")]
+        [InlineData(
+            "public const new int [|test|] = 1;",
+            "public const int [|test|] = 1;")]
+        [InlineData(
+            "public new event Action [|Test|];",
+            "public event Action Test;")]
+        [InlineData(
+            "public new int [|this[int p]|] => p;",
+            "public int this[int p] => p;")]
+        [InlineData(
+            "new class [|Test|] { }",
+            "class Test { }")]
+        [InlineData(
+            "new struct [|Test|] { }",
+            "struct Test { }")]
+        [InlineData(
+            "new interface [|Test|] { }",
+            "interface Test { }")]
+        [InlineData(
+            "new delegate [|Test|]()",
+            "delegate Test()")]
+        [InlineData(
+            "new enum [|Test|] { }",
+            "enum Test { }")]
+        [InlineData(
+            "new(int a, int b) [|test|];",
+            "(int a, int b) test;")]
+        public Task TestRemoveNewModifierFromMembersWithRegularFormatting(string original, string expected) =>
+            TestRemoveNewModifierCodeFixAsync(original, expected);
+
+        [Theory]
+        [InlineData(
+            "/* start */ public /* middle */ new /* end */ int [|Test|];",
+            "/* start */ public /* middle */ /* end */ int Test;")]
+        [InlineData(
+            "/* start */ public /* middle */ new    /* end */ int [|Test|];",
+            "/* start */ public /* middle */ /* end */ int Test;")]
+        [InlineData(
+            "/* start */ public /* middle */new /* end */ int [|Test|];",
+            "/* start */ public /* middle */ /* end */ int Test;")]
+        [InlineData(
+            "/* start */ public /* middle */ new/* end */ int [|Test|];",
+            "/* start */ public /* middle */ /* end */ int Test;")]
+        [InlineData(
+            "/* start */ public /* middle */new/* end */ int [|Test|];",
+            "/* start */ public /* middle *//* end */ int Test;")]
```

Comment:
fwiw, most of these tests are not realistic IMO.  i.e. even if they did appear, it would be a 0.001% case.  so exerting a ton of effort to make them work ok just isn't necessary.

Reply:
Ok, makes sense, can we move to this thread https://github.com/dotnet/roslyn/pull/43913#discussion_r419067063 for the discussion on what is not necessary to handle? (there is one strange failure there with `const`, for other cases there are just few extra spaces/line breaks)